startlxqtwayland.in

```diff
@@ -65,9 +65,29 @@ export QT_ACCESSIBILITY=1
 # use lxqt-applications.menu for main app menu
 export XDG_MENU_PREFIX="lxqt-"
 
+share_dir="$(dirname $(dirname "$0"))"/share
+
 if [ ! -d "$XDG_CONFIG_HOME/lxqt/wayland" ]; then
     mkdir -p $XDG_CONFIG_HOME/lxqt/wayland/
 fi
+# Check for default compositor and screenlocker eventually set by distribution
```

Comment:
IMO, this is wrong and does not follow the XDG standard.
1. no hard-coded path should be searched, but `XDG_CONFIG_DIRS` list, which is perfectly defined by the standard
2. user config file should not be modified anyhow here, as
2.1. the user hasn't made any decision here
2.2. the (distro default) value will be resolved anyways, when the configuration is needed and read by tools following the XDG standard

Ad 2.: The user just gets, what is the default value for his/her distro and if not actively making any preferences change, he/she gets the updated distro default value, when distro decides to change it. With the change made here, we break this expected scenario.

I know, I came late, but this PR is wrong and should be fixed ASAP.

Reply:
>2. user config file should not be modified anyhow here, as
2.1. the user hasn't made any decision here

He can't make any decision as the session would not even start without a setting for the compositor. For that the "firstrun" is tried (and works only if labws is present.)

>when distro decides to change it

that indeed is a point

>I know, I came late, but this PR is wrong and should be fixed ASAP.

Would you mind to make a PR or a draft? I can't see how we could display the compositor used in the settings atm if not `session.conf` is used.

Reply:
> He can't make any decision as the session would not even start without a setting for the compositor. For that the "firstrun" is tried (and works only if labws is present.)

Of course and therefore we must not write to user's config file, pretending he/she made the decision.

> I can't see how we could display the compositor used in the settings atm if not session.conf is used.

It is in the settings, either in the user's file or in file from the fallback directories. No matter where the value comes from, the "settings object" is aware of it (the Qt implementation didn't support searching all XDG_CONFIG_DIRS directories, but I my patch implementing it was accepted into Qt very very long time ago).

Reply:
>It is in the settings, either in the user's file or in file from the fallback directories. 

Oh, I wasn't aware at all of that - it's way more easier, something like 
```
if grep -q "compositor" "$share_dir/lxqt/session.conf"; then
    COMPOSITOR=`cat "$share_dir"/lxqt/session.conf|grep compositor |awk -F'=' '{sub(".*/", "", $2); print $2}'`
fi
echo "Compositor $COMPOSITOR set by distribution"
```
is enough then. But how can we use use `XDG_CONFIG_DIRS` instead of `$share_dir` here I don't see.